### ISSUE TITLE
Loosen python version requirement from `>=3.8,<4.0` to `>=3.8`

### ISSUE THREAD

**Issue by Ravencentric** [BrianPugh/cyclopts#200]
The python version requirement [here](https://github.com/BrianPugh/cyclopts/blob/20d2068d6c6356e2697e09be2c30d0656bf00e9b/pyproject.toml#L29) uses poetry's default `^` specifier which locks it down to `>=3.8,<4.0`. This means any project that declares their python version requirement like `>=3.8` cannot install it:

```
❯ poetry add cyclopts
Using version ^2.9.3 for cyclopts

Updating dependencies
Resolving dependencies... (0.0s)

The current project's supported Python range (>=3.8) is not compatible with some of the required packages Python requirement:
  - cyclopts requires Python <4.0,>=3.8, so it will not be satisfied for Python >=4.0

Because no versions of cyclopts match >2.9.3,<3.0.0
 and cyclopts (2.9.3) requires Python <4.0,>=3.8, cyclopts is forbidden.
So, because harbinger depends on cyclopts (^2.9.3), version solving failed.

  • Check your dependencies Python requirement: The Python requirement can be specified via the `python` or `markers` properties

    For cyclopts, a possible solution would be to set the `python` property to ">=3.8,<4.0"

    https://python-poetry.org/docs/dependency-specification/#python-restricted-dependencies,
    https://python-poetry.org/docs/dependency-specification/#using-environment-markers
```

**Comment by BrianPugh** [comment]
I got you, just a sec

**Comment by BrianPugh** [comment]
fixed in v2.9.4

**Comment by Ravencentric** [comment]
Thank you!
